exonum/src/helpers/fabric/clap_backend.rs

```diff
@@ -72,31 +72,22 @@ impl ClapBackend {
             .get_matches();
 
         let subcommand = matches.subcommand();
-        for command in commands {
-            if command.name() == subcommand.0 {
-                return command.execute(Context::new_from_args(
-                    command.args(),
-                    subcommand.1.expect("Arguments not found."),
-                ));
-            }
-        }
-
-        panic!("Subcommand not found");
+        let command = commands.get(subcommand.0).expect("Subcommand not found.");
+        command.execute(
+            commands,
+            Context::new_from_args(command.args(), subcommand.1.expect("Arguments not found.")),
+        )
     }
 
     fn command_into_subcommand(command: &CollectedCommand) -> clap::App {
-        let mut index = 1;
         let command_args: Vec<_> = command
             .args()
             .iter()
-            .map(|arg| {
+            .zip(1..)
+            .map(|(arg, index)| {
                 let clap_arg = clap::Arg::with_name(arg.name);
                 let clap_arg = match arg.argument_type {
-                    ArgumentType::Positional => {
-                        let arg = clap_arg.index(index);
-                        index += 1;
```

Comment:
Revert such way of indexing, as we have to count only positional arguments. Thanks, @vldm, for noticing